articles/quickstart/webapp/laravel/01-login.md

```diff
@@ -41,77 +60,154 @@ Optionally, if you want to use the [facade](http://laravel.com/docs/facades) cal
 
+class AppServiceProvider extends ServiceProvider
+{
+    // ...
+    public function register()
+    {
+        $this->app->bind(
+            Auth0Contract::class,
+            UserRepo::class
+        );
+    }
 }
 ```
 
 ## Configure It
 
-To configure the plugin, you need to publish the plugin configuration and complete the file `config/laravel-auth0.php` using the information of your Auth0 account.
+To configure the plugin, you must publish the plugin configuration and complete the file `config/laravel-auth0.php` using  information from your Auth0 account and details of your implementation.
 
-To publish the example configuration file use this command
+Publish the default configuration file with the following command:
 
 ```bash
 php artisan vendor:publish
 ```
 
-## Setup the Callback Action
+Select the option for `Auth0\Login\LoginServiceProvider` and look for `Publishing complete.` This creates the file `config/laravel-auth0.php` with the following settings:
 
-The plugin works with the [Laravel authentication system](https://laravel.com/docs/5.3/authentication), but instead of using the `Auth::attempt` in a controller that handles a login form submit, you have to hook up the callback uri.
+* `domain` - Your Auth0 tenant domain, found in your Client settings (required)
+* `client_id` - Your Auth0 Client ID, found in your Client settings (required)
+* `client_secret` - Your Auth0 Client Secret, found in your Client settings (required)
+* `redirect_uri` - The callback URI for your Laravel application to handle the login response from Auth0; by default this is `APP_URL/auth0/callback` (required)
+* `persist_user` - Should the user information persist in a PHP session? Default is `true`
+* `persist_access_token` - Should the Access Token persist in a PHP session? Default is `false`
+* `persist_id_token` - Should the ID Token persist in a PHP session? Default is `false`
+* `authorized_issuers` - An array of authorized token issuers, should include at least your tenant domain
+* `api_identifier` - The authorized token audience
+* `secret_base64_encoded` - Is the Client Secret Base64 encoded? Look below the Client Secret field in the Auth0 dashboard to see how to set this; default is `false`
+* `supported_algs` - JWT decoding algorithms supported by your application; default is `RS256`
+* `guzzle_options` - Specify additional [request options for Guzzle](http://docs.guzzlephp.org/en/stable/request-options.html)
 
-In other words, you need to select a uri (for example `/auth0/callback`) and configure it in your [Auth0 admin page](${manage_url}/#/applications) and also, add it as a route in `routes/web.php`:
+To keep sensitive data out of version control and allow for different testing and production instances, we recommend using the `.env` file Laravel uses to load other environment-specific variables:
+
+```
+// .env
+// ...
+AUTH0_DOMAIN=${account.namespace}
+AUTH0_CLIENT_ID=${account.clientId}
+AUTH0_CLIENT_SECRET=${account.clientSecret}
```

Comment:
This should just say `'YOUR_CLIENT_SECRET'`, as we're steering away from using this particular variable in docs, and I'd not even thought about that until it just came up in a conversation and I remembered this PR :)

Reply:
@jeffreylees - You're welcome to change it back :)